lib/hanami/view/escape.rb

```diff
@@ -165,10 +165,13 @@ def self.extended(base)
       # @since 0.4.0
       # @api private
       def method_added(method_name)
+        visibility = :private if private_method_defined? method_name
+        visibility ||= :protected if protected_method_defined? method_name
```

Comment:
Can you write some tests that leverage this distinction? They will help explain this, ensure there are no regressions in the future, and who knows... might even help you find a simpler solution :D